pgmigrate.sh

```diff
@@ -58,18 +121,27 @@ while [[ $# > 0 ]]; do
     esac
 done
 
-check_migrate
-if [[ "${RET_VALUE}" != 0 ]]; then
-    if [[ "${INSTALL_DEPENDENCY}" = 0 ]]; then
-        echoerr "Cannot find migrate. Provide \`--install-dependency\` to attempt to install it."
-        exit 1
-    fi
-    
-    go get -tags 'postgres' github.com/golang-migrate/migrate/v4/cmd/migrate
-    check_migrate
-    if [[ "${RET_VALUE}" != 0 ]]; then
-        echoerr "Failed to install migrate. Please install it manually."
-        exit 1
+check_migrate --verbose
+if [[ "${RET_VALUE}" != "OK" ]]; then
+    if [[ "${INSTALL_DEPENDENCY}" == 0 ]]; then
+        if [[ "${RET_VALUE}" == "NOT_FOUND" ]]; then
+            echoerr "Cannot find migrate. Provide \`--install-dependency\` to try to install it."
+            exit 1
+        fi
+    else
+        is_golang_1_16_and_above
+        if [[ "${RET_VALUE}" == 1 ]]; then
+#            install_golang_migrate_golang_1_16_and_above
+            install_golang_migrate_universal
```

Comment:
Looks like no difference between `if` and `else` clause as install_golang_migrate_universal is running `install_golang_migrate_golang_1_15_and_below`

Reply:
I guess `install_golang_migrate_golang_1_16_and_above` should be uncommented?

Reply:
This is intended. I tried `install_golang_migrate_golang_1_16_and_above` at the beginning but I realize the `migrate` installed using this method has `migrate --version` always returns `dev` so it confuses the version check.

However, I want to keep the code here for future reference because this is the recommended way.

Reply:
I see